lib/active_triples/relation.rb

```diff
@@ -153,8 +168,14 @@ def add_child_node(resource,object=nil)
         resource.persist! if resource.persistence_strategy.is_a? ParentStrategy
       end
 
+      ##
+      # Gives the predicate used by the Relation. Values of this object are 
+      # those that match the pattern `<rdf_subject> <predicate> [value] .`
+      #
+      # @return [RDF::Term] the predicate for this relation
       def predicate
-        property.kind_of?(RDF::URI) ? property : property_config[:predicate]
+        return property if property.is_a?(RDF::Term)
+        property_config[:predicate] unless property_config.nil?
```

Comment:
yeah, I suppose. Do you see that as a blocker here if I'm engaged in a major rework of `Relation`?